blog/_src/posts/2017-05-12-artifacts-for-semantics.md

```diff
@@ -0,0 +1,155 @@
+    Title: Artifacts for Semantics
+    Date: 2017-05-12T10:08:31
+    Tags: by Daniel Patterson
+
+[Gabriel Scherer](http://gallium.inria.fr/~scherer/) and I recently wrote
+an [artifact](https://dbp.io/artifacts/funtal) for a
+semantics [paper](https://dbp.io/pubs/2017/funtal.pdf) on a typed assembly
+language interoperating with a high-level functional language. Building
+artifacts to explore programming language semantics is nothing new for PRL --
+indeed, the PhD PL course at Northeastern is entirely taught
+using [Redex](https://redex.racket-lang.org/), but this post is about a
+different approach -- writing small-step interepreters in OCaml and compiling
+them to run on the web -- which has its own advantages.
+
+<!-- more -->
+
+[Redex](https://redex.racket-lang.org/) is a wonderful tool for exploring
+programming language semantics, but rather than use Redex, we wrote a
+interpreter, typechecker, and parser in OCaml, compiled it to Javascript
+using [js_of_ocaml](http://ocsigen.org/js_of_ocaml/), and then put it on a
+webpage (with an editor with syntax highlighting and error reporting) that
```

Comment:
There should be a link to the artifact here, and one in the header above as well (`Feel free to start by playing a bit with [our artifact](URL).`)